### PR TITLE
bug: BinaryOperatorSpacesFixer - fix for static in type

### PR DESCRIPTION
**Pull request by kubawerlos**
Fixes https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/6830

### CODE REVIEW

**Review by keradus — APPROVED** [review]

File: tests/Tokenizer/Transformer/TypeAlternationTransformerTest.php [hunk 1]
```diff
@@ -89,6 +89,32 @@ function foo(){}
...
+            [
+                12 => CT::T_TYPE_ALTERNATION,
+                14 => CT::T_TYPE_ALTERNATION,
+                34 => CT::T_TYPE_ALTERNATION,
+            ],
+        ];
+
+        yield 'static as type' => [
+            '<?php class Foo {
+                function f1(): static|TypeA {}
```
Comment (Wirone) [inline tests/Tokenizer/Transformer/TypeAlternationTransformerTest.php:107]:
```suggestion
                function f1(static|TypeA): void {}
```
Maybe like for `self`?

File: tests/Tokenizer/Transformer/TypeIntersectionTransformerTest.php [hunk 2]
```diff
@@ -342,6 +342,32 @@ function f2(FOO&BAR&BAZ&$x) {} // Intersection found
...
+            [
+                12 => CT::T_TYPE_INTERSECTION,
+                14 => CT::T_TYPE_INTERSECTION,
+                34 => CT::T_TYPE_INTERSECTION,
+            ],
+        ];
+
+        yield 'static as type' => [
+            '<?php class Foo {
+                function f1(): static&TypeA {}
```
Comment (Wirone) [inline tests/Tokenizer/Transformer/TypeIntersectionTransformerTest.php:360]:
```suggestion
                function f1(static&TypeA $x): void {}
```

**Review by Wirone — CHANGES_REQUESTED** [review]
Slight improvements - use unions and intersections both as argument's type and return types, both for `self` and `static`.

PS. Arrow functions 😝?

File: tests/Tokenizer/Transformer/TypeAlternationTransformerTest.php (hunk 1 above)
Comment (keradus) [reply tests/Tokenizer/Transformer/TypeAlternationTransformerTest.php:107]: we may want to test both, param type and return type

Comment (Wirone) [reply tests/Tokenizer/Transformer/TypeAlternationTransformerTest.php:107]: Sorry for the noise, I was wrong. `static` [can be only used as return type](https://wiki.php.net/rfc/static_return_type).

**Pull request merged**